contractcourt/channel_arbitrator.go

```diff
@@ -1744,9 +1744,15 @@ func (c *ChannelArbitrator) prepContractResolutions(
 					continue
 				}
 
+				circuitKey := channeldb.CircuitKey{
```

Comment:
w.r.t the commit message, I think replay awareness, rather than protection is more descriptive, as replay protection happens at the onion layer for HTLCs.

Reply:
The commit also combines recognizing these replays, with resolving some accounting level gaps related to HTLC acceptance. IMO they're better committed distinctly, packaged with tests for each change and a proper commit body that provides background/rationale, etc.

Reply:
Because we recognize the replays, the accounting gap is closed. One is almost a side effect of the other. I don't see how to separate this into two commits.

Fixed the commit message and also the messages on other commits. Those are definitely lacking in this pr and should have received more attention before review.

Reply:
Unit tests extended to cover improvements in this pr.